security/pfSense-pkg-snort/files/usr/local/www/snort/snort_alerts.php

```diff
@@ -423,62 +412,44 @@ function snort_match_filter_field($flent, $fields, $exact_match = FALSE) {
 	// action lists.
 	switch ($action) {
 		case "action_default":
-			if (isset($alertsid[$gid][$sid])) {
-				unset($alertsid[$gid][$sid]);
+			if (array_get_path($alertsid, "{$gid}/{$sid}")) {
+				array_del_path($alertsid, "{$gid}/{$sid}");
 			}
-			if (isset($dropsid[$gid][$sid])) {
-				unset($dropsid[$gid][$sid]);
+			if (array_get_path($dropsid, "{$gid}/{$sid}")) {
+				array_del_path($dropsid, "{$gid}/{$sid}");
 			}
-			if (isset($rejectsid[$gid][$sid])) {
-				unset($rejectsid[$gid][$sid]);
+			if (array_get_path($rejectsid, "{$gid}/{$sid}")) {
+				array_del_path($rejectsid, "{$gid}/{$sid}");
 			}
 			break;
 
 		case "action_alert":
-			if (!is_array($alertsid[$gid])) {
-				$alertsid[$gid] = array();
-			}
-			if (!is_array($alertsid[$gid][$sid])) {
-				$alertsid[$gid][$sid] = array();
+			array_set_path($alertsid, "{$gid}/{$sid}", "alertsid");
+			if (array_get_path($dropsid, "{$gid}/{$sid}")) {
+				array_del_path($dropsid, "{$gid}/{$sid}");
 			}
-			$alertsid[$gid][$sid] = "alertsid";
-			if (isset($dropsid[$gid][$sid])) {
-				unset($dropsid[$gid][$sid]);
-			}
-			if (isset($rejectsid[$gid][$sid])) {
-				unset($rejectsid[$gid][$sid]);
+			if (array_get_path($rejectsid, "{$gid}/{$sid}")) {
+				array_del_path($rejectsid, "{$gid}/{$sid}");
 			}
 			break;
 
 		case "action_drop":
-			if (!is_array($dropsid[$gid])) {
-				$dropsid[$gid] = array();
-			}
-			if (!is_array($dropsid[$gid][$sid])) {
-				$dropsid[$gid][$sid] = array();
-			}
-			$dropsid[$gid][$sid] = "dropsid";
-			if (isset($alertsid[$gid][$sid])) {
-				unset($alertsid[$gid][$sid]);
+			array_set_path($dropsid, "{$gid}/{$sid}", "dropsid");
+			if (array_get_path($alertsid, "{$gid}/{$sid}")) {
+				array_del_path($alertsid, "{$gid}/{$sid}");
 			}
-			if (isset($rejectsid[$gid][$sid])) {
-				unset($rejectsid[$gid][$sid]);
+			if (array_get_path($rejectsid, "{$gid}/{$sid}")) {
+				array_del_path($rejectsid, "{$gid}/{$sid}");
 			}
 			break;
 
 		case "action_reject":
-			if (!is_array($rejectsid[$gid])) {
-				$rejectsid[$gid] = array();
-			}
-			if (!is_array($rejectsid[$gid][$sid])) {
-				$rejectsid[$gid][$sid] = array();
-			}
-			$rejectsid[$gid][$sid] = "rejectsid";
-			if (isset($alertsid[$gid][$sid])) {
-				unset($alertsid[$gid][$sid]);
+			array_set_path($rejectsid, "{$gid}/{$sid}", "rejectsid");
+			if (array_get_path($alertsid, "{$gid}/{$sid}")) {
+				array_del_path($alertsid, "{$gid}/{$sid}");
 			}
-			if (isset($dropsid[$gid][$sid])) {
-				unset($dropsid[$gid][$sid]);
+			if (array_get_path($dropsid, "{$gid}/{$sid}")) {
+				array_del_path($dropsid, "{$gid}/{$sid}");
```

Comment:
You can safely array_del_path() without the conditional check, if the path does not exist no exception is thrown and no action is taken.

Reply:
Thanks. Implemented the suggested change, and it greatly simplifies the code.